src/libraries/Common/src/System/Net/SocketProtocolSupportPal.Windows.cs

```diff
@@ -5,67 +5,30 @@
 using System.Diagnostics;
 using System.Net.Sockets;
 using System.Runtime.InteropServices;
-using System.Threading;
 #if !SYSTEM_NET_SOCKETS_DLL
 using SocketType = System.Net.Internals.SocketType;
 #endif
 
 namespace System.Net
 {
-    internal class SocketProtocolSupportPal
+    internal static class SocketProtocolSupportPal
     {
-        private static bool s_ipv4 = true;
-        private static bool s_ipv6 = true;
-
-        private static bool s_initialized;
-        private static readonly object s_initializedLock = new object();
-
-        public static bool OSSupportsIPv6
-        {
-            get
-            {
-                EnsureInitialized();
-                return s_ipv6;
-            }
-        }
-
-        public static bool OSSupportsIPv4
-        {
-            get
-            {
-                EnsureInitialized();
-                return s_ipv4;
-            }
-        }
-
-        private static void EnsureInitialized()
-        {
-            if (!Volatile.Read(ref s_initialized))
-            {
-                lock (s_initializedLock)
-                {
-                    if (!s_initialized)
-                    {
-                        s_ipv4 = IsProtocolSupported(AddressFamily.InterNetwork);
-                        s_ipv6 = IsProtocolSupported(AddressFamily.InterNetworkV6);
-
-                        Volatile.Write(ref s_initialized, true);
-                    }
-                }
-            }
-        }
+        public static bool OSSupportsIPv6 { get; } = IsProtocolSupported(AddressFamily.InterNetworkV6);
+        public static bool OSSupportsIPv4 { get; } = IsProtocolSupported(AddressFamily.InterNetwork);
+        public static bool OSSupportsUnixDomainSockets { get; } = IsProtocolSupported(AddressFamily.Unix);
 
         private static bool IsProtocolSupported(AddressFamily af)
```

Comment:
Nitpick: Shouldn't these methods be called `IsAddressFamilySupported()`?  It's not checking if a protocol is supported by the underlying network API (it's in fact specified as 0 explicitly in the call to mean "auto-detect based on address family and socket type").

Reply:
Fixed the name